### PR TITLE
[Perf Framework]

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: #394 
fixes: #393 

Adding noOp and and another test to demo how a test can have and get its own test options.

This is working version of the performance framework based on .NET implementation for C++.

### CODE REVIEW

**Comment by vhvb1989** [comment]
Running the noOp test:

image

File: CMakeLists.txt [hunk 1]
```diff
@@ -64,8 +65,10 @@ include(AzureDoxygen)
 include(AzureVersion)
 
 # sub-projects
-# add_subdirectory(sdk/core/performance-stress)
 add_subdirectory(sdk/core/azure-core)
+if (BUILD_PERFORMANCE_TESTS)
```
Comment (RickWinter) [inline CMakeLists.txt:69]: This needs to go in the azure-core CMakeLists.txt.

File: sdk/core/performance-stress/CMakeLists.txt [hunk 2]
```diff
@@ -2,27 +2,37 @@
 # SPDX-License-Identifier: MIT
 
 cmake_minimum_required (VERSION 3.13)
-project(azure-perf-stress LANGUAGES CXX)
+project(azure-performance-stress LANGUAGES CXX)
```
Comment (RickWinter) [inline sdk/core/performance-stress/CMakeLists.txt:5]: Why is it "performance-stress" and not "perf-framework" or "performance-framework".

File: sdk/core/performance-stress/inc/azure/performance-stress/argagg.hpp [hunk 3]
```diff
@@ -0,0 +1,1541 @@
+/*
```
Comment (RickWinter) [inline sdk/core/performance-stress/inc/azure/performance-stress/argagg.hpp:1]: This is a lot of code for argument parsing.  Why not use a simple map<string, string>?

If you stay with this parser the liscense needs to be approved.

File: sdk/core/performance-stress/CMakeLists.txt (hunk 2 above)
Comment (vhvb1989) [reply sdk/core/performance-stress/CMakeLists.txt:5]: the name comes from the .Net version. We can change it, I think. I don´t think there are guidelines for it

File: sdk/core/performance-stress/inc/azure/performance-stress/argagg.hpp (hunk 3 above)
Comment (vhvb1989) [reply sdk/core/performance-stress/inc/azure/performance-stress/argagg.hpp:1]: This is for the application arguments from console line. The ones we get from the main(int, char **).  We can´t have a map there.

The license is MIT.

**Review by antkmsft — APPROVED** [review]

File: sdk/core/performance-stress/inc/azure/performance-stress/argagg.hpp (hunk 3 above)
Comment (RickWinter) [reply sdk/core/performance-stress/inc/azure/performance-stress/argagg.hpp:1]: @vhvb1989 This package needs to be approved before it can be used.

Comment (vhvb1989) [reply sdk/core/performance-stress/inc/azure/performance-stress/argagg.hpp:1]: MIT: https://github.com/vietjtnguyen/argagg

image

**Pull request merged**